mmdet/models/dense_heads/deformable_detr_head.py

```diff
@@ -38,20 +37,20 @@ class DeformableDETRHead(DETRHead):
 
     def __init__(self,
                  *args,
-                 with_box_refine: bool = False,
                  as_two_stage: bool = False,
-                 transformer: OptConfigType = None,
+                 with_box_refine: bool = False,
+                 num_decoder_layers: int = 6,
                  **kwargs) -> None:
-        self.with_box_refine = with_box_refine
+        # NOTE The three key word args are set in the detector,
+        # the users do not need to set them in config.
         self.as_two_stage = as_two_stage
-        if self.as_two_stage:
-            transformer['as_two_stage'] = self.as_two_stage
+        self.with_box_refine = with_box_refine
```

Comment:
Would renaming it as `share_pred_layers` be better? The prediction head's setting should be decoupled with the decoder layers.

Reply:
Yes, it is. I'll modify this arg. Thanks for your review.

Reply:
I have add the following tip to the docstring of four step (pre_transformer, forward_encoder, pre_decoder, forward_decoder) for the specific DETR detectors.

```
The forward procedure of the transformer is defined as:
'pre_transformer' -> 'encoder' -> 'pre_decoder' -> 'decoder'
More details can be found at `TransformerDetector.forward_transformer`
in `mmdet/detector/base_detr.py`.
```

Thanks for your great suggestion!